joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstForFunctionsCreator.scala

```diff
@@ -87,9 +87,11 @@ trait AstForFunctionsCreator(implicit withSchemaValidation: ValidationMode) { th
   protected def getReceiverInfo(receiver: Try[Value]): Option[(String, String, String, ParserNodeInfo)] = {
     receiver match
       case Success(rec) if rec != null =>
-        val recnode        = createParserNodeInfo(rec)
-        val recValue       = rec(ParserKeys.List).arr.head
-        val recName        = recValue(ParserKeys.Names).arr.head(ParserKeys.Name).str
+        val recnode  = createParserNodeInfo(rec)
+        val recValue = rec(ParserKeys.List).arr.head
```

Comment:
`.head` is not defensive nor is it in a `Try`, could you see if something like this works?

```scala
val recName = rec(ParserKeys.List)
  .arr
  .headOption
  .flatMap(recValue => 
    Try(recValue(ParserKeys.Names).arr.head(ParserKeys.Name).str).getOrElse(Defines.This)
  )
```

Reply:
Yea, I am working on handling that scenario along with few other use cases as well. Will handle that as part of next PR.